torchhd/utils.py

```diff
@@ -0,0 +1,63 @@
+import torchhd.functional as functional
+
+def plot_basis_set(A, ax=None, **kwargs):
+    """Displays a similarity map for the provided set of hypervectors as a 
+    matrix map
+
+    Args:
+        A: 2D array-like
+            The set of hypervectors whose similarity map is to be displayed
+        
+        ax: matplotlib Axes, optional
+        Axes in which to draw the plot
+        
+    Other Parameters:
+        **kwargs: `~matplotlib.pyplot.matshow` arguments
+
+    Returns:
+        ax: matplotlib.axes
+    """
+    try:
+        import matplotlib.pyplot as plt
+    except ImportError:
+        raise ImportError("Install matplotlib to use plotting functionality. \
+        See https://matplotlib.org/stable/users/installing/index.html for more information.")
+
+    sim = []
+    for vector in A:
+        sim.append(functional.cosine_similarity(vector, A).tolist())
+    if ax is None:
+        ax = plt.gca()
+    axes = ax.matshow(sim, **kwargs)
+    return axes
+
+def plot_vector_similarity(A, vec, ax=None, **kwargs):
```

Comment:
Maybe we can name this just `plot_similarity`? I think that would be clear enough. And again, we should use more descriptive argument names, `input` for `vec` and `memory` for `A` will make it more consistent with the rest of the library. Could you also swap the order of `vec` and `A`, I think it would read more clearly if it says "plot_similarity *of* input *with* memory".